core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -641,6 +643,28 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def sendChangeNotifications(workspaceName: WorkspaceName) = {
+
+    def getUserEmails = {
+      dataSource.inTransaction{ dataAccess =>
+        withWorkspaceContext(workspaceName, dataAccess) {workspaceContext =>
+          requireAccessIgnoreLock(workspaceContext.workspace, WorkspaceAccessLevels.Write, dataAccess) {
+            DBIO.sequence(workspaceContext.workspace.accessLevels.values.map { group =>
+              dataAccess.rawlsGroupQuery.flattenGroupMembership(group)
+            }.map(_.map(user => userInfo.userEmail)))
+          }
+        }
+      }
+    }
+    for {
+      emails <- getUserEmails
+    } yield {
+      val notificationMessages = emails.map(email => Notifications.WorkspaceChangedNotification(email, workspaceName))
+      notificationDAO.fireAndForgetNotifications(notificationMessages)
+    }
```

Comment:
this for comprehension could be rewritten as

```
    getUserEmails.map { emails =>
      val notificationMessages = emails.map(email => Notifications.WorkspaceChangedNotification(email, workspaceName))
      notificationDAO.fireAndForgetNotifications(notificationMessages)
    }
```

They are equivalent. Usually when there is only 1 element in the for I just go with map. For comprehensions are really handy when you start nesting flatMaps and maps.